deploy/templates/default-managed-identity-azuredeploy.json

```diff
@@ -409,7 +410,7 @@
     },
     {
       "type": "Microsoft.EventHub/namespaces/eventhubs/providers/roleAssignments",
-      "apiVersion": "2018-07-01",
+      "apiVersion": "2018-09-01-preview",
```

Comment:
Agreed we should avoid using preview APIs where we can

Reply:
For roleAssignments, I'm only seeing preview versions listed, and the one I used is the min version needed to fix the intermittent failures due to replication delay. Didn't want to update to latest preview in case it brought in some unwanted changes that we aren't ready for. But let me know if I am looking at the wrong place, or if its okay to update to the latest preview version listed.
https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/roleassignments?tabs=json

![image](https://user-images.githubusercontent.com/84752343/121092363-46b69e00-c7a0-11eb-8028-6012359c1cbb.png)

Reply:
I'm actually quite confused on how to interpret the versions to use for role assignments. We aren't going by the Microsoft.Authorization/roleAssignments in these templates, but using resourcetype/providers/roleAssignments instead because of scope issues. 
The only reason I updated it here because it seems like to address the "Principal doesn't exist" based on this https://docs.microsoft.com/en-us/azure/role-based-access-control/role-assignments-template#new-service-principal